src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -75,9 +77,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
     Entity lectureEntity = getLectureEntityFromRequest(request);
     datastore.put(lectureEntity);
+    initializeTranscript(lectureEntity);
     response.sendRedirect(buildRedirectUrl(lectureEntity).get());
   }
 
+  /**
+   * Parses and stores the transcript lines in datastore using the {@code lectureKey}
+   * and {@code videoId} properties in {@code lectureEntity}.
+   */
+  private void initializeTranscript(Entity lectureEntity) throws IOException, ServletException {
+    TranscriptServlet transcriptServlet = new TranscriptServlet();
```

Comment:
So I believe there's a way to get a reference to the existing servlet without having to create/init a new one.
It's something like this:
```
TranscriptServlet s = (TranscriptServlet) getServletContext().getServlet("transcript");
```

where "transcript" comes from the name given in the web.xml config

Reply:
Resolved. Thanks for you help! :)